java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c

```diff
@@ -639,33 +637,29 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addExe
   const char* epName = (*jniEnv)->GetStringUTFChars(jniEnv, jepName, NULL);
   const OrtApi* api = (const OrtApi*)apiHandle;
   OrtSessionOptions* options = (OrtSessionOptions*)optionsHandle;
-  size_t keyCount = (*jniEnv)->GetArrayLength(jniEnv, configKeyArr);
+  int keyCount = (*jniEnv)->GetArrayLength(jniEnv, configKeyArr);
 
-  if (keyCount > ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH) {
-    throwOrtException(jniEnv, ORT_INVALID_ARGUMENT, "Too many provider options.");
-  }
-
-  const char* keyArray[ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH];
-  const char* valueArray[ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH];
-  jstring jkeyArray[ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH];
-  jstring jvalueArray[ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH];
+  const char** keyArray = (const char**)allocarray(keyCount, sizeof(const char*));
+  const char** valueArray = (const char**)allocarray(keyCount, sizeof(const char*));
+  jstring* jkeyArray = (jstring*)allocarray(keyCount, sizeof(jstring));
+  jstring* jvalueArray = (jstring*)allocarray(keyCount, sizeof(jstring));
 
-  for (size_t i = 0; i < keyCount; i++) {
-    // In this loop, it is type-safe to call (jsize)i since i is
-    // upper-bounded by keyCount (type: jsize) returned by GetArrayLength.
-    jkeyArray[i] = (jstring)((*jniEnv)->GetObjectArrayElement(jniEnv, configKeyArr, (jsize)i));
-    jvalueArray[i] = (jstring)((*jniEnv)->GetObjectArrayElement(jniEnv, configValueArr, (jsize)i));
+  for (int i = 0; i < keyCount; i++) {
```

Comment:
Java's array cannot have more than 2G elements. Therefore int is enough. `size_t` is not needed here.